src/main/java/org/broadinstitute/hellbender/utils/variant/VariantContextVariantAdapter.java

```diff
@@ -24,9 +35,48 @@
     public boolean isSnp() { return variantContext.isSNP(); }
     @Override
     public boolean isIndel() { return variantContext.isIndel(); }
+
+    @Override
+    public UUID getUUID() {
+        return uuid;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
If we're really only using VariantContext for its Variant-related fields (and ignoring the rest), then this seems overly restrictive. The contract of "equals" is that it checks for value equality. If we have two different representations of the same variant, shouldn't equal return true? They would, after all, represent the same variant. 

This means we can just use VariantUtils.variantsAreEqual and remove most of this code (we could also reuse the hash method that way).